packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -112,6 +123,12 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 				change,
 			});
 
+			// If this transaction is not nested, add it to the undo commit tree
+			if (this.transactions.size === 0) {
+				commitRepairStore?.capture(this.changeFamily.intoDelta(change), this.head.revision);
```

Comment:
What the advantage of having `commitTransaction` perform the capture instead of passing undefined to `this.undoRedoManager.trackCommit` ?

Reply:
The commitRepairStore has a reference to a clone of the forest from before any changes were applied as part of this transaction while passing undefined would mean that undoRedoManager creates a repair store using its repairDataStoreFactory which just references the forest.